tensorflow/python/data/experimental/ops/get_single_element.py

```diff
@@ -20,31 +20,175 @@
-  def preprocessing_fn(input_str):
-    # ...
-    return image, label
+  def preprocessing_fn(raw_feature):
+    # ... the raw_feature is preprocessed as per the use-case
+    return feature
 
-  input_batch = ...  # input batch of BATCH_SIZE elements
-  dataset = (tf.data.Dataset.from_tensor_slices(input_batch)
+  raw_features = ...  # input batch of BATCH_SIZE elements.
+  dataset = (tf.data.Dataset.from_tensor_slices(raw_features)
              .map(preprocessing_fn, num_parallel_calls=BATCH_SIZE)
              .batch(BATCH_SIZE))
 
-  image_batch, label_batch = tf.data.experimental.get_single_element(dataset)
+  processed_features = tf.data.get_single_element(dataset)
+  ```
+
+  In the above example, the `raw_features` tensor of length=BATCH_SIZE
+  was converted to a `tf.data.Dataset`. Next, each of the `raw_feature` was
+  mapped using the `preprocessing_fn` and the processed features were
+  grouped into a single batch. The final `dataset` contains only one element
+  which is a batch of all the processed features.
+
+  NOTE: The `dataset` should contain only one element.
+
+  Now, instead of creating an iterator for the `dataset` and retrieving the
+  batch of features, the `tf.data.get_single_element()` function is used
+  to skip the iterator creation process and directly output the batch of
+  features.
+
+  This can be particularly useful when your tensor transformations are
+  expressed as `tf.data.Dataset` operations, and you want to use those
+  transformations while serving your model. Especially, when it comes to
+  serving `tf.keras` models using a `SavedModel`.
+
+  tf.keras
+  --------
+
+  ```python
+  import tensorflow as tf
+
+  # Load data
+  (x_train, y_train), (x_test, y_test) = tf.keras.datasets.fashion_mnist.load_data()
+  CLASSES = [
+    "T-shirt/top", "Trouser", "Pullover", "Dress", "Coat", "Sandal",
+    "Shirt", "Sneaker", "Bag", "Ankle Boot"
+  ]
+
+  def make_dataset(images, labels):
+    ds = tf.data.Dataset.from_tensor_slices((images, labels))
+    ds = ds.shuffle(buffer_size=1024)
+    ds = ds.map(lambda x, y: (x / 255, y))
+    return ds.batch(batch_size=128)
+
+  def make_model():
+    # Add the layers
+    model = tf.keras.Sequential([
+      tf.keras.layers.Flatten(),
+      tf.keras.layers.Dense(10)
+    ])
+    # Compile the model
+    model.compile( optimizer='adam', metrics=['accuracy'],
+      loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+    )
+    return model
+
+  model = make_model()
+  train_data = make_dataset(x_train, y_train)
+  test_data = make_dataset(x_test, y_test)
+  model.fit(train_data, validation_data=test_data, epochs=10)
+
+  # There is a need to pre-process the data before inference as the model
+  # will throw exceptions if the input is not of the desired shape. Thus, to
+  # serialize and save the pre-processing functionality along with the model,
+  # a `PreprocessingModel` is defined with a custom serving function called
+  # `serving_fn`.
```

Comment:
Simplify this as follows:

```
model = ...

class PreprocessingModel(tf.keras.Model):
  def __init__(self, model):
    super().__init__(self)
    self.model = model

  @tf.function(input_signature=[...])
  def serving_fn(self, data):
    ds = tf.data.Dataset.from_tensor_slices(data)
    ds = ds.map(preprocessing_fn, num_parallel_calls=BATCH_SIZE)
    ds = ds.batch(batch_size=BATCH_SIZE)
    return tf.argmax(self.model(tf.data.experimental.get_single_element(ds)), axis=-1)

preprocessing_model = PreprocessingModel(model)
tf.saved_model.save(preprocessing_model, your_exported_model_path,
                signatures={'serving_default': preprocessing_model.serving_fn})
```

Doing this will provide consistency between the tf.keras and tf.estimator example.

Reply:
addressed in the commit.

Reply:
Please update the body of `serving_fn` as per my suggestion.

Reply:
@kvignesh1420 this still needs to be addressed. Thanks.

Reply:
@jsimsa sorry about that. I addressed it in the commit.